### PR TITLE
Bazel6 - stamping tars, generating gh-page.js, and deploying gh-pages

### PR DESCRIPTION
**Pull request by mercillo**
This PR does 3 things;

It takes a tar and stamps it with any substitutions.

On rules_player build, it will also generate a compiled gh-pages.js script from the gh-pages.ts file.

this allows us to be able to call gh_pages to publish that page

### CODE REVIEW

File: distribution/tar/stamp_tar_files.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const fs = require("fs");
+const path = require("path");
+const os = require('os');
+const cp = require('child_process')
+const process = require("process")
+// const tar = require('tar');
```
Comment (KetanReddy) [inline distribution/tar/stamp_tar_files.js:6]: I would just delete this line

File: distribution/tar/stamp_tar_files.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+const fs = require("fs");
+const path = require("path");
+const os = require('os');
+const cp = require('child_process')
+const process = require("process")
```
Comment (adierkens) [inline distribution/tar/stamp_tar_files.js:5]: Can we update this to use `.mjs` and the ES module syntax (`import`)?

File: distribution/tar/BUILD [hunk 3]
```diff
@@ -0,0 +1,17 @@
+load("@aspect_rules_js//js:defs.bzl", "js_binary")
+load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
+
+bzl_library(
```
Comment (adierkens) [inline distribution/tar/BUILD:4]: Make sure to add `distribution` to the list of docs to generate

Comment (adierkens) [reply distribution/tar/BUILD:4]: Also the formatting/linting rules: 

```
load("@cgrindel_bazel_starlib//bzlformat:defs.bzl", "bzlformat_missing_pkgs", "bzlformat_pkg")

bzlformat_pkg(name = "bzlformat")

bzlformat_missing_pkgs(name = "bzlformat_missing_pkgs")
```

File: package.json [hunk 4]
```diff
@@ -9,7 +9,23 @@
     "type": "git",
     "url": "https://github.com/player-ui/rules_player.git"
   },
+  "scripts": {
+    "build:ts": "yarn esbuild --bundle --platform=node gh-pages/gh-pages.ts --outfile=gh-pages/gh-pages.js"
```
Comment (adierkens) [inline package.json:13]: Is there a reason this isn't included as part of a bazel build rule?

File: .circleci/config.yml [hunk 5]
```diff
@@ -59,6 +62,15 @@ jobs:
...
             - .
 
+  build_ts:
+    executor: base
+    steps:
+      - attach_workspace:
+          at: ~/rules_player
+      - run:
+          name: Test
+          command: yarn build:ts
```
Comment (adierkens) [inline .circleci/config.yml:72]:
```suggestion
          command: bazel build //...
```

File: package.json (hunk 4 above)
Comment (adierkens) [reply package.json:13]: Does this path map to anything? It looks like the script is under `gh-pages/private/gh-pages.ts`

**Comment by KetanReddy** [comment]
Should the bundled `.js` file be tracked in vcs if its going to get built during rules build time?

**Comment by mercillo** [comment]
> Should the bundled `.js` file be tracked in vcs if its going to get built during rules build time?

there was a test that was failing because it was trying to find a .js file

**Review by hborawski — APPROVED** [review]

**Comment by intuit-svc** [comment]
:rocket: PR was released in [`v1.1.0`](https://github.com/player-ui/rules_player/releases/tag/v1.1.0) :rocket:

**Pull request merged**